lib/LogPriceLib.sol

```diff
@@ -52,11 +53,24 @@ library LogPriceLib {
   }
 
   function outboundFromInboundUp(int logPrice, uint inboundAmt) internal pure returns (uint) {
-    (uint sig, uint exp) = LogPriceConversionLib.nonNormalizedPriceFromLogPrice(-logPrice);
-    uint prod = sig*inboundAmt;
-    return (prod>>exp) + (prod%(1<<exp)==0 ? 0 : 1);
+    unchecked {
+      (uint sig, uint exp) = LogPriceConversionLib.nonNormalizedPriceFromLogPrice(-logPrice);
+      return divExpUp(sig*inboundAmt,exp);
+    }
   }
+}
 
-
-
+// Return a/2**e rounded up
+function divExpUp(uint a, uint e) pure returns (uint) {
+  unchecked {
+    uint carry;
+    /* 
+    Easier to represent carry in assembly.
+    carry is 1 if a&mask > 0, where mask is (1<<e)-1, and carry is 0 otherwise
+    */
+    assembly("memory-safe") {
+      carry := gt(and(a,sub(shl(e,1),1)),0)
+    }
```

Comment:
I'm convinced, but I needed the explanation you gave me on Slack, so maybe include it here?

Copying it here for convenience :-)

> if a == 0then a & x == 0 for all x, so carry is 0
> otherwise, if e > 255 then (1<<e)-1 == type(uint).max so a & ((1<<e)-1) == a > 0, so carry is 1 (as it should, since 0 < a < 2^e)
> otherwise we’re in the normal case since a & ((1<<e)-1) == a % (1<<e)

Reply:
done in https://github.com/mangrovedao/mangrove-core/pull/553/commits/32333eaabfaae5a7f65ed6867aec611f88701c41